mutation/index.ts

```diff
@@ -0,0 +1,132 @@
+import { useCallback, useRef } from 'react'
+import useSWR, {
+  useSWRConfig,
+  Middleware,
+  Key,
+  Fetcher,
+  SWRResponse
+} from 'swr'
+
+import { serialize } from '../src/utils/serialize'
+import { useStateWithDeps } from '../src/utils/state'
+import { withMiddleware } from '../src/utils/with-middleware'
+import { useIsomorphicLayoutEffect } from '../src/utils/env'
+import { isUndefined, UNDEFINED } from '../src/utils/helper'
+import { getTimestamp } from '../src/utils/timestamp'
+
+import { SWRMutationConfiguration, SWRMutationResponse } from './types'
+
+const mutation = <Data, Error>() => (
+  key: Key,
+  fetcher: Fetcher<Data>,
```

Comment:
```suggestion
  fetcher: Required<Fetcher<Data>>,
```

Reply:
It’s still possible that the fetcher comes from the config arg or context.